packages/webcomponents/src/ui-components/form/iframe-input.tsx

```diff
@@ -0,0 +1,110 @@
+import { Component, h, Method, Prop, State } from "@stencil/core";
+import { FrameCommunicationService } from "../../utils/frame-comunication-service";
+import { StyledHost } from "../styled-host/styled-host";
+import { MessageEventType } from "../../components/payment-method-form/message-event-types";
+import { FormControlErrorText } from "./form-helpers/form-control-error-text";
+import packageJson from '../../../package.json';
+
+@Component({
+  tag: "iframe-input",
+  shadow: true,
+})
+export class IframeInput {
+  private iframeElement!: HTMLIFrameElement;
+  private frameService: FrameCommunicationService;
+
+  @State() isFocused: boolean = false;
+  @State() isValid: boolean = true;
+  @State() errorText: string;
+
+  @Prop() inputId: string;
+  @Prop() label: string;
+  @Prop() iframeOrigin: string;
+
+  disconnectedCallback() {
+    this.frameService.removeMessageListener(this.dispatchMessageEvent);
+  }
+
+  @Method()
+  async validate(): Promise<any> {
+    const response = await this.frameService
+      .postMessageWithResponseListener(
+        MessageEventType.card.validate
+      );
+
+    this.isValid = response.isValid;
+
+    this.errorText = response.error.message;
+
+    return response;
+  }
+
+  @Method()
+  async tokenize(
+    clientId: string,
+    paymentMethodMetadata: any,
+    account?: string,
+  ) {
+    const eventType = MessageEventType['card'].tokenize;
+    return this.frameService.postMessageWithResponseListener(eventType, {
+      clientId,
+      paymentMethodMetadata,
+      account,
+      componentVersion: packageJson.version,
+    });
+  }
+
+  private initializeFrameCommunicationService() {
+    this.frameService = new FrameCommunicationService(
+      this.iframeElement, this.iframeOrigin
+    );
+    this.frameService?.addMessageListener(this.dispatchMessageEvent);
+  }
+
+  private dispatchMessageEvent = (event: MessageEvent) => {
+    const { eventType, data } = event.data;
+    if (data?.id !== this.inputId) {
+      return;
+    }
+
+    if (eventType === 'focused') {
+      this.isFocused = true;
+    }
+
+    if (eventType === 'blurred') {
+      this.isFocused = false;
+    }
+  }
+
+  private get fontFamily() {
+    return btoa(getComputedStyle(document.body).fontFamily);
```

Comment:
Nice! I love that we can get this programatically. Are we able to specifically get the computed styles of an `input` with `form-control` class? It wouldn't be the cleanest, but since this is used along-side other inputs maybe we can compute the font family off of a native input at the parent level and pass it down to these `frame-input`s?

Reply:
Nice, the suggestion of adding a hidden input with bootstrap styles worked fine!